apps/pkcs12.c

```diff
@@ -60,58 +60,73 @@ typedef enum OPTION_choice {
     OPT_R_ENUM
 } OPTION_CHOICE;
 
+typedef enum OPTION_group {
+    OPT_GRP_GENERAL = 0,
+    OPT_GRP_PASSWORD,
+    OPT_GRP_ENCRYPT,
+    OPT_GRP_MAC,
+    OPT_GRP_CA,
+    OPT_GRP_ATTR,
+    OPT_GRP_INPUT,
+    OPT_GRP_OUTPUT,
+} OPTION_GROUP;
+
+const char* pkcs12_opt_groups[] = {
+    "General", "Password", "Encryption", "MAC", "CA Certificate", "Attribute", "Input", "Output"
+};
```

Comment:
I would very much prefer if there was a `struct { int groupnum; const char *groupname }` for the elements of this array, so it could be given like this:

``` C
struct GROUPS pkcs12_groups[] = {
    { OPT_GRP_GENERAL, "General" },
    { OPT_GRP_PASSWORD, "Password" },
    { OPT_GRP_ENCRYPT, "Encryption" },
    { OPT_GRP_MAC, "MAC" },
    { OPT_GRP_CA, "CA Certificate" },
    { OPT_GRP_ATTR, "Attribute" },
    { OPT_GRP_INPUT, "Input" },
    { OPT_GRP_OUTPUT,  "Output" },
};
```

(incidently, there's already a type you can use rather than defining `GROUP`, called `OSSL_ITEM`, found in `include/openssl/core.h`)

Having to synchronise the positions in an array with a separate enum is a maintenance problem waiting to happen.  We've had that in the past, which is why I'm pushing for something that makes the connection more explicit.